### PR TITLE
33792 - Fix Director Display for COOP AR

### PR DESCRIPTION
**Pull request by meawong**
*Issue #:* /bcgov/entity#33792 

*Description of changes:*
- Update logic to handle non-null cessation dates for future-ceased directors in COOP AR
- Add/update unit tests
- App version = 8.3.18

Ex Before:
image

Ex Now:
image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-815-w20s1aj4.web.app

Example Test Setup:
- Jia Xu CEASED on February 1, 2024
- Megan Wong APPOINTED on February 1, 2024
- John Smith APPOINTED on March 1, 2024

Compare Dev:
https://dev.business.bcregistry.gov.bc.ca/CP1002597/annual-report?accountid=3040&businessid=CP1002597&filingId=0&arFilingYear=2024

With Preview: https://business-filings-dev--pr-815-w20s1aj4.web.app/CP1002597/annual-report?accountid=3040&businessid=CP1002597&filingId=0&arFilingYear=2024

File: tests/unit/AnnualReport.spec.ts [hunk 1]
```diff
@@ -1586,7 +1586,7 @@ describe('Annual Report - Part 5 - Data', () => {
             addressRegion: 'BC'
           },
           appointmentDate: '2019-01-01',
-          cessationDate: '2019-03-25',
+          cessationDate: '2017-06-15',
```
Comment (kialj876) [inline tests/unit/AnnualReport.spec.ts:1589]: why this change? Wouldn't this be invalid?

**Review by kialj876 — APPROVED** [review]
Looks good to me - just one question below to confirm

File: src/components/common/Directors.vue [hunk 2]
```diff
@@ -756,7 +756,7 @@ export default class Directors extends Mixins(CommonMixin, DateMixin, DirectorMi
           this.original.forEach((director, i) => {
             director.id = i + 1
             director.isFeeApplied = (director.isFeeApplied !== undefined) ? director.isFeeApplied : false
-            director.isDirectorActionable = !director.cessationDate
+            director.isDirectorActionable = !director.cessationDate || director.cessationDate > this.asOfDate
```
Comment (mruff-aeq) [inline src/components/common/Directors.vue:762]: I noticed the `cessationDate` checks differ between the two files, which might lead to edge-case bugs down the road. Specifically, `src/components/common/Directors.vue:759` uses `!director.cessationDate`, while `src/views/AnnualReport.vue:1156` uses `el.cessationDate === null`.

If `cessationDate` ever comes through as `''` or `undefined` instead of `null`, these components will behave differently. **`Directors.vue`** will treat the director as active because `!cessationDate` evaluates to `true`. Conversely, **`AnnualReport.vue`** will drop them from the filing payload because `cessationDate === null` evaluates to `false`.

This means a director visible in the UI could be silently missing from what is submitted in the edge case. @kialj876 am i over thinking things? 

`director.isDirectorActionable = director.cessationDate === null || director.cessationDate > this.asOfDate`

File: tests/unit/AnnualReport.spec.ts (hunk 1 above)
Comment (meawong) [reply tests/unit/AnnualReport.spec.ts:1589]: Forgot to update the `appointmentDate` as well, I've fixed this and thanks for flagging.

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-815-w20s1aj4.web.app

File: src/components/common/Directors.vue (hunk 2 above)
Comment (meawong) [reply src/components/common/Directors.vue:762]: I've updated this check to align with `Directors.vue`

Comment (kialj876) [reply src/components/common/Directors.vue:762]: Yeah this is a good point, but I think we can let this PR through as it'll fix a current prod bug and this is an existing issue with inconsistency in the code. I think we can probably wait to update this once its upgraded and moved into our nuxt layers as the filing will be pretty heavily refactored as part of that

Comment (kialj876) [reply src/components/common/Directors.vue:762]: Oh right on 👍

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-815-w20s1aj4.web.app

**Review by mruff-aeq — APPROVED** [review]

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-filings-dev--pr-815-w20s1aj4.web.app
